tests/test_url.py

```diff
@@ -647,6 +647,14 @@ def test_with_query():
     assert str(url.with_query({'a': '1'})) == 'http://example.com/?a=1'
 
 
+def test_with_query_kwargs():
+    url = URL('http://example.com')
+    assert str(url.with_query(
+        {'a': '2', 'b': '4'}, a='1')) == 'http://example.com/?a=1'
+    assert 'query2=1' in str(url.with_query(query='1', query2='1'))
```

Comment:
@asvetlov because kwargs doesn't have ordering, I'm not sure what order variables will be in the end. So I'm not sure how url will look like in the end. is this wrong ?

It looks like

```
http://example.com/?query2=1&query=1
```

But I'm not sure why, on other machines it can look ```http://example.com/?query=1&query2=1```